### PR TITLE
For #18175 - Add a new total protection ETP cookies policy

### LINKED ISSUE #18175: Total cookie protection
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have exactly one feature request described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Feature Requests are better when they’re open-ended instead of demanding a specific solution e.g: “I want an easier way to do X” instead of “add Y”)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

### What is the user problem or growth opportunity you want to see solved?
Recently Mozilla has added total cookie protection in v86 however it seems this is not yet added or implemented in Firefox Android. Please add the same

### How do you know that this problem exists today? Why is this important?
Total cookie protection missing 

### Who will benefit from it?
All users

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-9819)

### PR DESCRIPTION
**Pull request by Mugurell**
This should implement the UX from https://www.figma.com/file/OUOpOzgd4EOdgwJZvBGQ9j/TCP?node-id=286%3A13848

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

Fixes #18175
Fixes #18175

### CODE REVIEW

**Comment by Amejia481** [comment]
Thanks for the patch 👏🏽 , there is a dependency that we need to address before landing on Fenix, as wihtout it we will introduce some breakage to Fenix see https://github.com/mozilla-mobile/focus-android/issues/6027 and  https://github.com/mozilla-mobile/android-components/issues/10739

File: app/src/main/res/values/arrays.xml [hunk 1]
```diff
@@ -2,20 +2,23 @@
 <resources>
 
     <string name="social">social</string>
+    <string name="total_protection">total-protection</string>
```
Comment (Amejia481) [inline app/src/main/res/values/arrays.xml:5]: To follow desktop settings we should make this our default :)

**Comment by Mugurell** [comment]
Waiting for https://github.com/mozilla-mobile/android-components/pull/11399 to be merged.

**Comment by topotropic** [comment]
Do you still need input here? Thanks!

**Comment by Mugurell** [comment]
> Do you still need input here? Thanks!

Was interested if the cookies blocking policy - "Cross-site cookies" (Firefox's Total protection) seems okay for Fenix and 
whether we should show it as the first option in the custom ETP settings. 
(Focus and desktop which already have this option are not really in sync)

**Comment by topotropic** [comment]
@Mugurell yes, that looks alright, thanks!

**Comment by Mugurell** [comment]
Rebased this patch to integrate the recent changes about the new storage_access permission.

**Review by Amejia481 — CHANGES_REQUESTED** [review]
After chatting with the privacy team, they communicate us that we only have to enable TCP on strict mode, we should keep our existing default value for standard mode. After this change I think we can land patch :)

**Comment by Amejia481** [comment]
> After chatting with the privacy team, they communicate us that we only have to enable TCP on strict mode and private mode, we should keep our existing default value for standard mode. After this change I think we can land patch :)

To keep this setting consistent across browsers, I'll put a patch on AC, then here we will only have to remove the part that changes our default value.

**Comment by Amejia481** [comment]
> > After chatting with the privacy team, they communicate us that we only have to enable TCP on strict mode and private mode, we should keep our existing default value for standard mode. After this change I think we can land patch :)
> 
> To keep this setting consistent across browsers, I'll put a patch on AC, then here we will only have to remove the part that changes our default value.

The PR  https://github.com/mozilla-mobile/android-components/pull/11481

**Comment by gabrielluong** [comment]
Please review with @athomasmoz before landing.

**Comment by Mugurell** [comment]
> Will this PR bring feature parity between Firefox Desktop and Fenix regarding ETP Strict?

Final specs are still in progress.

**Comment by opusforlife2** [comment]
Specs finalised, then, since you're updating the PR?

**Comment by Mugurell** [comment]
Tried updating this but I couldn't.
Will open a new PR.

**Comment by opusforlife2** [comment]
@Mugurell The event stated "This PR was closed". It seems to occur when the source repo/branch gets deleted.

**Pull request closed without merging**